### ISSUE TITLE
Will it handle also SPOILER ?

### ISSUE THREAD

**Issue by xcom169** [speed47/spectre-meltdown-checker#273]
Will it handle also SPOILER defect?

**Comment by knweiss** [comment]
Paper: [SPOILER: Speculative Load Hazards Boost Rowhammer and Cache Attacks](https://arxiv.org/pdf/1903.00446.pdf) (pdf)

**Comment by MarvinOl** [comment]
Microsoft has released Windows 10 Insider Preview 18860.1001 that includes 16 new microcodes for Intel CPUs in mcupdate_GenuineIntel.dll as well as SPOILER workarounds in the Windows Kernel.

I extracted the microcode for my CPU and included it to Linux for loading and the register changes were:
old microcode:  eax in: 0x00000007, ... edx = 2c000000
new microcode: eax in: 0x00000007, ... edx = 2c000400

old microcode: IA32_ARCH_CAPS=0x19
new microcode:IA32_ARCH_CAPS=0x79

So at the moment Intel is setting both undocumented EDX and IA32_ARCH_CAPS bits for SPOILER fix.

**Comment by MarvinOl** [comment]
Microsoft has released Windows 10 Insider Preview 18860.1001 that includes 16 new microcodes for Intel CPUs in mcupdate_GenuineIntel.dll as well as SPOILER workarounds in the Windows Kernel.

The new microcodes can be found in https://github.com/platomav/CPUMicrocodes and included to Linux for loading. For my CPU the register changes were:
old microcode: eax in: 0x00000007, ... edx = 2c000000
new microcode: eax in: 0x00000007, ... edx = 2c000400

old microcode: IA32_ARCH_CAPS=0x19
new microcode: IA32_ARCH_CAPS=0x79

So at the moment Intel is setting both undocumented EDX and IA32_ARCH_CAPS bits for SPOILER fix.

**Comment by speed47** [comment]
Those flags are interesting indeed, even if we have no official information yet that they are SPOILER-related.
I'll probably add checks for them at least in a dev branch for the time being, thanks @MarvinOl !

**Comment by MarvinOl** [comment]
New details about the new exploit and fixes:
https://access.redhat.com/security/vulnerabilities/mds
https://portal.msrc.microsoft.com/en-us/security-guidance/advisory/adv190013

**Comment by speed47** [comment]
Yep. First detection version is already merged, I'm still currently refining it, `master` will be noisy tonight.
Btw, it wasn't SPOILER after all.

**Comment by speed47** [comment]
AFAICT, there is no SPOILER mitigation on OS or firmware level, as it's just a way to enhance a Rowhammer attack. So, there's no existing mitigation for the script to detect. Leaving this open for information only.
